### PR TITLE
Loot action

### PR DESCRIPTION
**Pull request by ryan3r**

### CODE REVIEW

File: src/main/java/pro/trevor/tankgame/rule/impl/shared/PlayerRules.java [hunk 1]
```diff
@@ -289,13 +377,21 @@ public static PlayerConditionRule spendActionToShootWithDeathHandle(
                 }
             });
 
+    public static final PlayerConditionRule SHOOT_V5 = spendActionToShootWithDeathHandle(
+        LineOfSight::hasLineOfSightV4,
+        (s, tank, dead) -> {
```
Comment (tygard) [inline src/main/java/pro/trevor/tankgame/rule/impl/shared/PlayerRules.java:382]: make these better

File: src/main/java/pro/trevor/tankgame/rule/impl/shared/PlayerRules.java [hunk 2]
```diff
@@ -214,14 +224,92 @@ public static PlayerConditionRule getRuleCofferCostBounty(int lowerBound, int up
...
+            // Make sure that the position is an AttributeObject
+            Position position = toType(meta[0], Position.class);
+            Optional<IElement> targetElement = state.getBoard().getUnitOrFloor(position);
+            if(targetElement.isEmpty()) {
+                return Result.error("Target " + position + " is out of bounds");
+            }
+
+            if(!(targetElement.get() instanceof AttributeObject)) {
+                return Result.error("Cannot target " + position + " it's the wrong type");
+            }
```
Comment (tygard) [inline src/main/java/pro/trevor/tankgame/rule/impl/shared/PlayerRules.java:249]: make better add incorrect class in error message

File: src/main/java/pro/trevor/tankgame/state/board/Board.java [hunk 3]
```diff
@@ -125,7 +125,7 @@ public Optional<IPlayerElement> getPlayerElement(PlayerRef playerRef) {
     // If there is no unit at the position, then the floor is returned.
     public Optional<IElement> getUnitOrFloor(Position p) {
         IElement unit = getUnit(p).orElse(null);
-        if (!(unit instanceof EmptyUnit))
+        if (unit != null && !(unit instanceof EmptyUnit))
             return Optional.of(unit);
         IElement floor = getFloor(p).orElse(null);
```
Comment (TrevorBrunette) [inline src/main/java/pro/trevor/tankgame/state/board/Board.java:130]: This can just be `return getFloor(p)`

**Review by TrevorBrunette — CHANGES_REQUESTED** [review]

File: src/main/java/pro/trevor/tankgame/state/board/Board.java (hunk 3 above)
Comment (ryan3r) [reply src/main/java/pro/trevor/tankgame/state/board/Board.java:130]: I tried that java can't "convert" from `Optional<IFloor>` to `Optional<IElement>`

> Type mismatch: cannot convert from Optional<IFloor> to Optional<IElement>

**Review by TrevorBrunette — APPROVED** [review]

**Review by tygard — APPROVED** [review]

**Pull request merged**